src/Library/demos/Revealer/main.js

```diff
@@ -0,0 +1,14 @@
+const button = workbench.builder.get_object("button");
+const revealer = workbench.builder.get_object("revealer");
+
+button.connect("clicked", () => {
+  revealer.reveal_child = !revealer.reveal_child;
+  switch (button.label) {
+    case "Show":
+      button.label = "Hide";
+      return;
+    case "Hide":
+      button.label = "Show";
+      return;
```

Comment:
never use user facing strings for conditions, it's fragile

you can use the state of the toggle button